protocols/request-response/src/handler.rs

```diff
@@ -384,19 +402,47 @@ where
         loop {
             match self.worker_streams.poll_unpin(cx) {
                 Poll::Ready((_, Ok(Ok(event)))) => {
-                    return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(event))
+                    return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(event));
                 }
                 Poll::Ready((id, Ok(Err(e)))) => {
-                    log::debug!("Stream for request {id} failed: {e}");
+                    let event = match id {
+                        RequestId::Inbound(id) => {
+                            log::debug!("Stream for inbound request {id} failed: {e}");
+                            Event::InboundStreamFailed {
+                                request_id: id,
+                                error: e,
+                            }
+                        }
+                        RequestId::Outbound(id) => {
+                            log::debug!("Stream for outbound request {id} failed: {e}");
+                            Event::OutboundStreamFailed {
+                                request_id: id,
+                                error: e,
+                            }
+                        }
+                    };
+
+                    return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(event));
                 }
                 Poll::Ready((id, Err(futures_bounded::Timeout { .. }))) => {
-                    log::debug!("Stream for request {id} timed out");
+                    let event = match id {
+                        RequestId::Inbound(id) => {
+                            log::debug!("Stream for inbound request {id} timed out");
+                            Event::InboundTimeout(id)
+                        }
+                        RequestId::Outbound(id) => {
+                            log::debug!("Stream for outbound request {id} timed out");
+                            Event::OutboundTimeout(id)
+                        }
+                    };
+
+                    return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(event));
```

Comment:
Same here, I'd collapse the pattern match and remove the logs.

Reply:
done